### ISSUE TITLE
permanent logical replication slot can cause data loss

### ISSUE THREAD

**Issue by DeathBorn** [zalando/patroni#1056]
patroni 1.5.3
```
patronictl show-config
....
slots:
  permanent_logical_slot_metadata_history:
    database: metadata
    plugin: pgoutput
    type: logical
...
```
Setup:
- Patroni Cluster A with 2 nodes (leader Alice, sync. Standby Bob)
- Patroni Cluster B with 1 node (leader John Doe)
- permanent_logical_slot_C for Table X from cluster A -> cluster B using consul service discovery DNS address

Scenario
1. Cluster B goes down
2. Some time passes with new data about new Tesla car appearing in Table X
3. Cluster A leader failover to Bob
4. Cluster B brought back

Results:
During step 2, Alice sees that `permanent_logical_slot_C` is not active and accumulates wal files.
Bob is fast and he replicates everything. He discards old wals because he applied them.
After step 3, Bob recreates `permanent_logical_slot_C` (surprise surprise) with a latest LSN as if he never new about `permanent_logical_slot_C` existence on Alice side.
After step 4, John Doe starts consuming data from Bob.  **He has no data about Tesla car.**

Thoughts:
So basically, we can only rely only on wal_keep_segments being big enough and bringing John Doe ASAP. fixing missing data can be quite a hard task.

**Comment by CyberDem0n** [comment]
This is a known problem and there is no solution to it.

1. When you create a replication slot it is not possible to specify `restart_lsn`.
2. It is not possible to create logical replication slot on the replica.

Patroni is doing it's best to minimize the data loss. It "doesn't allow" connections (GET /master will not return 200, callbacks are not executed) until it makes an attempt to create logical slots.
If the consumer was connected to the slot during failover/switchover and it was not falling too much behind, with very high chances Patroni will create the slot on the new primary before some data was modified.

What we actually need from postgres, is a "failover slots" feature. It was discussed in the pgsql-hacker and there even were some patches, but unfortunately it is still not there.

**Comment by DeathBorn** [comment]
Thanks for confirmation. I saw some info regarding "failover slots" and thought maybe somebody already did something to solve this differently but seems not the case.

**Comment by perwigren** [comment]
How about creating the permanent slots on all nodes, then regularly call `pg_replication_slot_advance()` (pg11+) to forward the older ones to the newest restart_lsn of the slots?

With physical slots this would also make it possible to point for example Barman/pg_receivewal to the sync replica instead of the master, and setting up cascaded extra replicas from an async replica for reports, batch exports, etc.

**Comment by ants** [comment]
Currently you can't create logical slots on the standby, so that does not work.
